src/OpenLoco/Windows/ScenarioSelect.cpp

```diff
@@ -0,0 +1,503 @@
+            const int16_t y = self->y + self->widgets[widx::panel].top + 5;
+
+            auto str = const_cast<char*>(StringManager::getString(StringIds::buffer_2039));
+            strncpy(str, scenarioInfo->scenarioName, std::size(scenarioInfo->scenarioName));
+
+            auto args = FormatArguments();
+            args.push(StringIds::buffer_2039);
+            Gfx::drawStringCentredClipped(*dpi, x, y, 170, Colour::black, StringIds::wcolour2_stringid, &args);
+        }
+
+        // Outline for preview image
+        {
+            const int16_t x = self->x + self->widgets[widx::list].right + 24;
+            const int16_t y = self->y + self->widgets[widx::panel].top + 19;
+
+            Gfx::drawRectInset(dpi, x, y, 130, 130, self->colours[1], 0x30);
+        }
+
+        // Preview image?
+        if (scenarioInfo->hasFlag(ScenarioIndexFlags::hasPreviewImage))
+        {
+            const int16_t x = self->x + self->widgets[widx::list].right + 25;
+            const int16_t y = self->y + self->widgets[widx::panel].top + 20;
+
+            const auto imageId = 0;
+            const auto g1 = Gfx::getG1Element(imageId);
+            if (g1 != nullptr)
+            {
+                // Temporarily substitute a G1 image with the data in the scenario index
+                const auto backupG1 = *g1;
+                *g1 = {};
+                g1->offset = reinterpret_cast<uint8_t*>(scenarioInfo->preview);
+                g1->width = 128;
+                g1->height = 128;
+
+                // Draw preview image and restore original G1 image.
+                Gfx::drawImage(dpi, x, y, imageId);
+                *g1 = backupG1;
+
+                // Draw compass
+                Gfx::drawImage(dpi, x, y, ImageIds::height_map_compass);
+            }
+        }
+        else
+        {
+            int16_t x = self->x + self->widgets[widx::list].right + 25;
+            int16_t y = self->y + self->widgets[widx::panel].top + 21;
+
+            // No preview image -- a placeholder will have to do.
+            auto image = Gfx::recolour(ImageIds::random_map_watermark, self->colours[1]);
+            Gfx::drawImage(dpi, x, y, image);
+
+            x += 64;
+            y += 59;
+
+            auto args = FormatArguments();
+            args.push(StringIds::randomly_generated_landscape);
+
+            // Overlay random map note.
+            auto origin = Gfx::point_t(x, y);
+            Gfx::drawStringCentredWrapped(dpi, &origin, 128, Colour::black, StringIds::wcolour2_stringid, &args);
+        }
+
+        {
+            const int16_t x = self->x + self->widgets[widx::list].right + 4;
+            int16_t y = self->y + self->widgets[widx::panel].top + 155;
+
+            // Description
+            auto str = const_cast<char*>(StringManager::getString(StringIds::buffer_2039));
+            strncpy(str, scenarioInfo->description, std::size(scenarioInfo->description));
```

Comment:
Wouldn't it make more sense to limit the size to the length of `buffer_2039` (well, I guess ideally the lesser of them)?

Reply:
You're right. In this case, `scenarioInfo->description` is the lesser of them. Unfortunately, there's no direct way of getting the length of the 2039 buffer, so I opted for `scenarioInfo->description`'s length directly. Thoughts?